tests/manage/pv_services/test_delete_provisioner_pod_while_thick_provisioning.py

```diff
@@ -0,0 +1,138 @@
+from ocs_ci.helpers import helpers, disruption_helpers
+
+
+logger = logging.getLogger(__name__)
+DISRUPTION_OPS = disruption_helpers.Disruptions()
+
+
+@tier4
+@tier4a
+class TestDeleteProvisionerPodWhileThickProvisioning(ManageTest):
+    """
+    Test to delete rbd provisioner leader pod while thick provisioning is progressing
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, project_factory):
+        """
+        Create Project for the test
+
+        Returns:
+            OCP: An OCP instance of project
+        """
+        self.proj_obj = project_factory()
+
+    @polarion_id("OCS-2531")
+    @bugzilla("1961647")
+    def test_delete_provisioner_pod_while_thick_provisioning(
+        self,
+        pvc_factory,
+        pod_factory,
+    ):
+        """
+        Test to delete RBD provisioner leader pod while creating a PVC using thick provision enabled storage class
+        """
+        pvc_size = 20
+        executor = ThreadPoolExecutor(max_workers=1)
+        DISRUPTION_OPS.set_resource(
+            resource="rbdplugin_provisioner", leader_type="provisioner"
+        )
+
+        # Start creation of PVC
+        pvc_create = executor.submit(
+            pvc_factory,
+            interface=constants.CEPHBLOCKPOOL,
+            project=self.proj_obj,
+            storageclass=default_thick_storage_class(),
+            size=pvc_size,
+            access_mode=constants.ACCESS_MODE_RWO,
+            status="",
+        )
+
+        # Ensure that the PVC is being created before deleting the rbd provisioner pod
+        ret = helpers.wait_for_resource_count_change(
+            get_all_pvcs, 0, self.proj_obj.namespace, "increase"
+        )
+        assert ret, "Wait timeout: PVC is not being created."
+        logger.info("PVC creation has started.")
+        DISRUPTION_OPS.delete_resource()
+        logger.info("Deleted RBD provisioner leader pod.")
+
+        pvc_obj = pvc_create.result()
+
+        # Confirm that the PVC is Bound
+        helpers.wait_for_resource_state(
+            resource=pvc_obj, state=constants.STATUS_BOUND, timeout=600
+        )
+        pvc_obj.reload()
+        logger.info(f"Verified: PVC {pvc_obj.name} reached Bound state.")
+
+        # Verify thick provision
+        pv_obj = pvc_obj.backed_pv_obj
+        image_name = pv_obj.get()["spec"]["csi"]["volumeAttributes"]["imageName"]
+        ct_pod = get_ceph_tools_pod()
+        du_out = ct_pod.exec_ceph_cmd(
+            ceph_cmd=f"rbd du -p {constants.DEFAULT_BLOCKPOOL} {image_name}",
+            format="",
+        )
+        used_size = "".join(du_out.strip().split()[-2:])
+        assert used_size == f"{pvc_size}GiB", (
+            f"PVC {pvc_obj.name} is not thick provisioned. Rbd image {image_name} expected used size: "
+            f"{pvc_size}GiB. Actual used size {used_size}.\n Rbd du out: {du_out}"
+            f"\n PV describe :\n {pv_obj.describe()}"
+        )
+        logger.info("Verified: The PVC is thick provisioned")
```

Comment:
can we create common function like is _thick_provisoned or get_provision type? we need to check this for all tick provisioned test cases.

Reply:
Created a helper function